packages/unified-selection/src/unified-selection/HiliteSetProvider.ts

```diff
@@ -53,10 +53,12 @@ class HiliteSetProviderImpl implements HiliteSetProvider {
   private _imodelAccess: ECClassHierarchyInspector & ECSqlQueryExecutor;
   // Map between a class name and its type
   private _classRelationCache: Map<string, InstanceIdType | Promise<InstanceIdType>>;
+  private _blockHandler: MainThreadBlockHandler;
 
   constructor(props: HiliteSetProviderProps) {
     this._imodelAccess = props.imodelAccess;
     this._classRelationCache = new Map();
+    this._blockHandler = new MainThreadBlockHandler({});
```

Comment:
Similar to my suggestions in `SelectionScope.ts`:
- I don't think we need to create the block handler here and pass it to `executeQuery` - the function can create its own block handler.
- I suggest having a local `executeQuery` that provides a row parser.

Reply:
Applied the suggestions.